engine-rest/engine-rest/src/test/java/org/camunda/bpm/engine/rest/PersistenceExceptionTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.camunda.bpm.engine.rest;
+
+import static io.restassured.RestAssured.given;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.camunda.bpm.engine.impl.util.ExceptionUtil.PERSISTENCE_EXCEPTION_MESSAGE;
+import static org.camunda.bpm.engine.impl.util.ExceptionUtil.wrapPersistenceException;
+import static org.camunda.bpm.engine.rest.exception.ExceptionLogger.REST_API;
+import static org.camunda.bpm.engine.rest.helper.MockProvider.EXAMPLE_USER_FIRST_NAME;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import java.sql.SQLNonTransientConnectionException;
+import java.util.List;
+import org.apache.ibatis.exceptions.PersistenceException;
+import org.camunda.bpm.engine.ProcessEnginePersistenceException;
+import org.camunda.bpm.engine.identity.UserQuery;
+import org.camunda.bpm.engine.rest.util.container.TestContainerRule;
+import org.camunda.commons.testing.ProcessEngineLoggingRule;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class PersistenceExceptionTest extends AbstractRestServiceTest {
+
+  protected static final String USER_QUERY_URL = TEST_RESOURCE_ROOT_PATH + "/user";
+
+  @ClassRule
+  public static TestContainerRule rule = new TestContainerRule();
+
+  @Rule
+  public ProcessEngineLoggingRule loggingRule = new ProcessEngineLoggingRule()
+      .watch(REST_API);
+
+  @Test
+  public void shouldLogWarning() {
```

Comment:
I've added also a more descriptive name for the case of persistence connection exceptions

Reply:
Thank you, much better already! What about the class names? The test for persistence connection exceptions is called `PersistenceConnectionTest`, and the one for non-persistence (connection??) exception is called `PersistenceExceptionTest`. Maybe we can clarify it even more?

Reply:
Pick a name for replacing `PersistenceExceptionTest` and i'm fine with replace it

Reply:
I don't know the technical details of your implementation, so this might not be correct. I would suggest to rename both classes:

* `PersistenceConnectionTest` -> `PersistenceConnectionExceptionLoggingTest`
* `PersistenceExceptionTest` -> `NonPersistenceConnectionExceptionLoggingTest` (or would it be `NonPersistenceExceptionLoggingTest`?)

Reply:
I've used the `PersistenceConnectionExceptionLoggingTest` & `NonPersistenceConnectionExceptionLoggingTest`. Also added a comment to explain the specific test cases for each class.